server/chat-plugins/announcements.ts

```diff
@@ -34,44 +33,40 @@ export class Announcement extends MinorActivity {
 
 	displayTo(user: User, connection: Connection | null = null) {
 		const recipient = connection || user;
-		recipient.sendTo(this.room, `|uhtml|announcement${this.announcementNumber}|${this.generateAnnouncement()}`);
+		recipient.sendTo(this.room, `|uhtml|announcement${this.activityNumber}|${this.generateAnnouncement()}`);
 	}
 
 	display() {
 		const announcement = this.generateAnnouncement();
 		for (const id in this.room.users) {
 			const thisUser = this.room.users[id];
-			thisUser.sendTo(this.room, `|uhtml|announcement${this.announcementNumber}|${announcement}`);
+			thisUser.sendTo(this.room, `|uhtml|announcement${this.activityNumber}|${announcement}`);
 		}
 	}
 
 	onConnect(user: User, connection: Connection | null = null) {
 		this.displayTo(user, connection);
 	}
 
-	end() {
-		this.endTimer();
-		this.room.send(`|uhtmlchange|announcement${this.announcementNumber}|<div class="infobox">(${this.room.tr`The announcement has ended.`})</div>`);
-		delete this.room.settings.minorActivity;
-		this.room.minorActivity = null;
-		this.room.saveSettings();
+	endActivity() {
```

Comment:
Rename to `destroy`? More consistent with PS APIs.

Reply:
sure